engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/PreviewWorldScreen.java

```diff
@@ -115,13 +135,63 @@ public void onOpened() {
     }
 
     @Override
+    public void update(float delta) {
+         super.update(delta);
+
+         if (triggerUpdate) {
+             updatePreview();
+             triggerUpdate = false;
+         }
+    }
+
+    private void configureProperties() {
+
+        PropertyLayout properties = find("properties", PropertyLayout.class);
+        properties.setOrdering(PropertyOrdering.byLabel());
+        properties.clear();
+
+        WorldConfigurator worldConfig = worldGenerator.getConfigurator();
+
+        Map<String, Component> params = worldConfig.getProperties();
+
+        for (String key : params.keySet()) {
+            Class<? extends Component> clazz = params.get(key).getClass();
+            Component comp = config.getModuleConfig(worldGenerator.getUri(), key, clazz);
+            if (comp != null) {
+                worldConfig.setProperty(key, comp);       // use the data from the config instead of defaults
+            }
+        }
+
+        for (String label : params.keySet()) {
+            PropertyProvider<?> provider = new PropertyProvider<>(params.get(label));
+            properties.addPropertyProvider(label, provider);
+        }
+    }
+
+    @Override
     public void onClosed() {
-        super.onClosed();
+
+        CoreRegistry.remove(WorldGeneratorPluginLibrary.class);
```

Comment:
Agreed. This change may also benefit from the gestalt-asset-core integration - ideally the preview screen should use a separate asset manager and environment from the main game, so that it doesn't leak assets. So it makes sense it should have its own context.